apps/wallet/src/generated/station/station.did

```diff
@@ -2083,6 +2150,169 @@ type CanisterStatusResponse = record {
+  // The time at which the canister was created.
+  created_at : TimestampRFC3339;
+  // The time at which the canister was last modified, if available.
+  modified_at : opt TimestampRFC3339;
+};
+
+// Input type for getting a external canister.
+type GetExternalCanisterInput = record {
+  // The principal id of the external canister.
+  canister_id : principal;
+};
+
+// The caller privileges for the external canister methods.
+type ExternalCanisterCallerMethodsPrivileges = record {
+  // The validation method that is used to validate the request and
+  // render the argument.
+  validation_method : ValidationMethodResourceTarget;
+  // The method that the caller can call on the external canister.
+  //
+  // The `*` method name is used to represent that the caller can
+  // call any method on the canister.
+  execution_method : text;
+};
+
+// The caller privileges for the external canister.
+type ExternalCanisterCallerPrivileges = record {
+  // The external canister retource id.
+  id : UUID;
+  // Wether or not the caller can edit the external canister.
+  can_change : bool;
+  // The list of methods that the caller can call on the external canister.
+  can_call : vec ExternalCanisterCallerMethodsPrivileges;
+};
+
+type GetExternalCanisterResult = variant {
+  Ok : record {
+    // The external canister that was retrieved.
+    canister : ExternalCanister;
+    // The caller privileges for the external canister.
+    privileges : ExternalCanisterCallerPrivileges;
+  };
+  Err : Error;
+};
+
+// Input type for listing external canisters with the given filters.
+type ListExternalCanistersInput = record {
+  // The principal id of the external canister to search for.
+  canister_ids : opt vec principal;
+  // The labels to use for filtering the external canisters.
+  labels : opt vec text;
+  // The pagination parameters.
+  paginate : opt PaginationInput;
+};
+
+// Result type for listing external canisters.
+type ListExternalCanistersResult = variant {
+  // The result data for a successful execution.
+  Ok : record {
+    // The list of external canisters.
+    canisters : vec ExternalCanister;
+    // The offset to use for the next page.
+    next_offset : opt nat64;
+    // The total number of external canisters.
+    total : nat64;
+    // The caller privileges for the external canisters.
+    privileges : vec ExternalCanisterCallerPrivileges;
+  };
+  // The error that occurred (e.g. the user does not have the necessary permissions).
+  Err : Error;
+};
+
+// The input type for getting the available filters for the external canisters.
+type GetExternalCanisterFiltersInput = record {
+  // Include the names of the external canisters in the result.
+  with_name : opt record {
+    // The prefix to use for filtering the names.
+    //
+    // If the prefix is not provided, any name will be returned.
+    prefix : opt text;
+  };
+  // Include the labels of the external canisters in the result.
+  with_labels : opt bool;
```

Comment:
any reason to use `opt bool` instead of just `bool`? Whats the difference between none and false?

Reply:
adding and removing `opt` fields is always safe, so, this enables clients of this api that would not be providing the field to still work since the field is optional, same goes if we decide to remove the field in the future.

and this would make it consistent if we decide to add more fields to the record without breaking clients.

Reply:
👍